### PR TITLE
DCS-1742 logic for new endpoint to get appointments information from db instead of calling NOMIS  

### PR DESCRIPTION
**Pull request by michalnawrockidj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/dto/VideoLinkBookingResponse.kt [hunk 1]
```diff
@@ -23,7 +23,7 @@ data class VideoLinkBookingResponse(
     example = "York Crown Court",
     required = true,
   )
-  val court: String,
+  val court: String?,
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/dto/VideoLinkBookingResponse.kt:26]: good spot!

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/VideoLinkBookingService.kt [hunk 2]
```diff
@@ -283,6 +322,11 @@ class VideoLinkBookingService(
       )
     }
 
+  private fun hasAnEndDate(it: VideoLinkAppointment): Boolean {
+    val hasEndDate = it.endDateTime != null
+    if (!hasEndDate) log.error("Appointment with id ${it.id} has no end date")
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/VideoLinkBookingService.kt:327]: is this possible? isn't end date a nullable field now?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/integration/CourtIntegrationTest.kt [hunk 3]
```diff
@@ -294,6 +294,105 @@ class CourtIntegrationTest(
...
+VALUES (2, 1182547, 'Wimbledon', 'WMBLMC', false, null, '2', null);"""
+      )
+      jdbcTemplate.update(
+        """INSERT INTO video_link_appointment (id, appointment_id, hearing_type, video_link_booking_id, location_id,
+                                           start_date_time, end_date_time)
+VALUES (4, 438577491, 'MAIN',2, 1234L, '2020-12-25 08:00:00', '2020-12-25 09:00:00');"""
+      )
+    }
+
+    fun deleteAll() {
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/integration/CourtIntegrationTest.kt:392]: should this just be in a beforeEach ?

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/VideoLinkBookingService.kt (hunk 2 above)
Comment (michalnawrockidj) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/VideoLinkBookingService.kt:327]: good spot. Fixed now

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**